src/components/ViewRequest/EditDrawer.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  Button,
+  Drawer,
+  DrawerBody,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerHeaderTitle,
+  Spinner,
+} from "@fluentui/react-components";
+import { DismissRegular } from "@fluentui/react-icons";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import * as FormFields from "components/Request/FormFields/FormFields";
+import { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { RPARequest, useMutateRequest, useRequest } from "api/requestsApi";
+import { SaveIcon } from "@fluentui/react-icons-mdl2";
+
+const EditDrawer = ({
+  isOpen,
+  setIsOpen,
+  subform,
+}: {
+  isOpen: boolean;
+  setIsOpen: (open: boolean) => void;
+  subform: string;
+}) => {
+  const params = useParams();
+  const request = useRequest(Number(params.requestId));
+  const updateRequest = useMutateRequest();
+
+  const myForm = useForm<RPARequest>({
+    criteriaMode:
+      "all" /* Pass back multiple errors, so we can prioritize which one(s) to show */,
+    mode: "onChange" /* Provide input directly as they input, so if entering bad data (eg letter in MPCN) it will let them know */,
+  });
+
+  useEffect(() => {
+    if (request.data) {
+      // Remove Author and Created properties from data object
+      const { Author, Created, ...data } = request.data ?? {};
+      myForm.reset(data);
+    }
+  }, [request.data]);
+
+  const onSubmit: SubmitHandler<RPARequest> = (data) => {
+    console.log(data);
+    updateRequest.mutateAsync(data);
+  };
+
+  return (
+    <Drawer
+      type="overlay"
+      position="end"
+      size="medium"
+      style={{ height: "100vh", minWidth: "fit-content" }}
+      open={isOpen}
+      onOpenChange={(_e, { open }) => setIsOpen(open)}
```

Comment:
Might be good to reset the fields upon open, so that if they open, make a change, cancel it, then reopen, the form resets to the values that are saved, and doesn't retain any unsaved edits.

Reply:
Should "isOpen" be added to the If statement, so that it only resets the form on open, and not on both open and close?